# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: ed71fdc2d78d
# Full Hash: ed71fdc2d78dec085a3d153220bba4aeda89992b
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-03-05 04:28:59
# Regressor Bug: 1951379
# File Overlap Count: 1
# Description:
#   Bug 1951379 - Part 3: Add NotificationStorageEntry r=asuth
#   
#   This skips handling `vibrate` for now, as we don't ship it and it also complicates test setup.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240213
# ==============================================================================

diff -r 61ab7888abc4 -r ed71fdc2d78d dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Tue Mar 04 18:00:17 2025 +0000
+++ b/dom/notification/NotificationParent.cpp	Tue Mar 04 18:00:18 2025 +0000
@@ -46,7 +46,8 @@
     (void)NS_WARN_IF(NS_FAILED(
         AdjustPushQuota(mPrincipal, NotificationStatusChange::Shown)));
     // XXX(krosylight): Non-persistent notifications probably don't need this
-    nsresult rv = PersistNotification(mPrincipal, mId, mOptions, mScope);
+    nsresult rv =
+        PersistNotification(mPrincipal, IPCNotification(mId, mOptions), mScope);
     if (NS_FAILED(rv)) {
       NS_WARNING("Could not persist Notification");
     }